examples/src/main/scala/org/apache/spark/examples/ml/DataFrameExample.scala

```diff
@@ -54,14 +54,13 @@ object DataFrameExample {
       }
     }
 
-    parser.parse(args, defaultParams).map { params =>
-      run(params)
-    }.getOrElse {
-      sys.exit(1)
+    parser.parse(args, defaultParams) match {
+      case Some(params) => run(params)
+      case _ => sys.exit(1)
     }
```

Comment:
It appears that you are reading what you want to see instead of what is really there.  All of the examples in the Scala API doc involve side-effects.  The three examples are essentially equivalent in their intent and effects, and using pattern matching with `Option` is clearly called out as the least idiomatic.

`fold` isn't in question since it has already been ruled out by other Spark committers as beyond the ken of those unfamiliar with idiomatic functional programming.  I see no reason, however, to do anything different in Spark's code than what the Scala API declares to be "most idiomatic way".